src/render/painter.js

```diff
@@ -430,9 +431,22 @@ class Painter {
         }
 
         if (this.options.showTileBoundaries) {
-            for (const id in sourceCaches) {
-                draw.debug(this, sourceCaches[id], coordsAscending[id]);
-                break;
+            //Use source with highest maxzoom
+            let selectedSource;
+            let sourceCache;
+            const layers = values(this.style._layers);
+            layers.forEach((layer) => {
```

Comment:
nit: Use a `for..of`?

Also would it be possible to clean up the loop somehow? seems its a bit unnecessarily complex for what it needs to do

Reply:
Simplified it a good bit 👍

Reply:
Looks like this has become moot as we've reverted to the original `forEach`, but for future reference just noting here that IE can't handle `for...of` :grimacing: